🦴 README skeleton (#13)
EthWorks/useDapp: README.md

# useDapp

Framework for creating Dapp on Ethereum with React. 
Easy. Secure. Testable. Extendable.

## Challange 

Writing Dapps comes with its own set of challenges and good practices:

- refresh UI after each new bock arrival
- work in two modes: read-only (without wallet connected) and read/write (with wallet connected)
- keep track of transactions status and inform a user when it is successfully mined or fails
- work with different wallets (Metamask, Formatic, Authereum) and networks (mainnet and testnets) and handle change nicely

With all that in mind, we want our Dapps to be easy to test and extend, build with high-pace development and security in mind.

## Solution

**useDapp** utilises best **React**, **Ethereum** and **testing** practices:
- use react hooks to communicate with the blockchain
- components automatically refresh on a new block
- multiple contract calls to are combined into a single multi-call
- extend with custom hooks
- write integration tests for components and blockchain with ease

## Example

```ts
function showBalance() {
  const {account} = useEthers();

>   const balance = useTokenBalance(account.address);

Reviewer: Either change to `useEthBalance` or specify a token address

Author: ✅

>   return (<EthersProvider>

Reviewer: Doesn't make sense: `useEthers` hook should be used in a component rendered inside `EthersProvider`

Author: ✅

# useDapp

Framework for creating Dapp on Ethereum with React. 
Easy. Secure. Testable. Extendable.

## Challange 

Writing Dapps comes with its own set of challenges and good practices:

- refresh UI after each new block arrival
- work in two modes: read-only (without wallet connected) and read/write (with wallet connected)
- keep track of transactions status and inform a user when it is successfully mined or fails
- work with different wallets (Metamask, Formatic, Authereum) and networks (mainnet and testnets) and handle these changes nicely

With all that in mind, we want our Dapps to be easy to test and extend, built with high-pace development and security in mind.

## Solution

**useDapp** utilises best **React**, **Ethereum** and **testing** practices:
- use react hooks to communicate with the blockchain
- components automatically refresh on a new block

> - multiple contract calls to are combined into a single multi-call

Reviewer 2: ```suggestion
- multiple contract calls are combined into a single multi-call
```

- extend with custom hooks
- write integration tests for components and blockchain with ease

## Example

> ```ts

Reviewer 2: ```suggestion
```tsx
```

Reviewer 2: This will colour code better :)

# useDapp

Framework for creating Dapp on Ethereum with React.
Easy. Secure. Testable. Extendable.

> ## Challange 

Reviewer 3: ```suggestion
## Challenge
```

Writing Dapps comes with its own set of challenges and good practices:

- refresh UI after each new block arrival
- work in two modes: read-only (without wallet connected) and read/write (with wallet connected)
- keep track of transactions status and inform a user when it is successfully mined or fails
- work with different wallets (Metamask, Formatic, Authereum) and networks (mainnet and testnets) and handle these changes nicely

With all that in mind, we want our Dapps to be easy to test and extend, built with high-pace development and security in mind.

## Solution

**useDapp** utilises best **React**, **Ethereum** and **testing** practices:

- use react hooks to communicate with the blockchain
- components automatically refresh on a new block
- multiple contract calls are combined into a single multi-call
- extend with custom hooks
- write integration tests for components and blockchain with ease

## Example

```tsx
ReactDOM.render(
  <EthersProvider>
    <App />

>   <EthersProvider/>

Reviewer 3: ```suggestion
  <EthersProvider/>,
```

  document.getElementById('app')
)

function App() {
  const balance = useEtherBalance();
  return (
    <div>
    Your balance is: <span> {balance} </span>
    </div>

>     )

Reviewer: ```suggestion
  )
```

# useDapp

Framework for creating Dapp on Ethereum with React. 
Easy. Secure. Testable. Extendable.

## Challange 

Writing Dapps comes with its own set of challenges and good practices:

> - refresh UI after each new bock arrival

Reviewer 2: ```suggestion
- refresh UI after each new block arrival
```

- work in two modes: read-only (without wallet connected) and read/write (with wallet connected)
- keep track of transactions status and inform a user when it is successfully mined or fails

> - work with different wallets (Metamask, Formatic, Authereum) and networks (mainnet and testnets) and handle change nicely

Reviewer 2: ```suggestion
- work with different wallets (Metamask, Formatic, Authereum) and networks (mainnet and testnets) and handle these changes nicely
```

> With all that in mind, we want our Dapps to be easy to test and extend, build with high-pace development and security in mind.

Reviewer 2: ```suggestion
With all that in mind, we want our Dapps to be easy to test and extend, built with high-pace development and security in mind.
```
